test/onnx/verify_onnx.cpp

```diff
@@ -1266,6 +1266,77 @@ TEST_CASE(resize_downsample_f_test)
     EXPECT(migraphx::verify::verify_range(result_vector, gold));
 }
 
+TEST_CASE(resize_downsample_f_dyn_test)
+{
+    migraphx::onnx_options options;
+    options.default_dyn_dim_value = {1, 10};
+    options.use_dyn_output        = true;
+
+    auto p = migraphx::parse_onnx("resize_downsample_f_dyn_test.onnx", options);
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape sx{migraphx::shape::float_type, {2, 1, 5, 9}};
+    std::vector<float> dx(sx.elements());
+    std::iota(dx.begin(), dx.end(), 0.1f);
+
+    migraphx::parameter_map pp;
+    pp["X"] = migraphx::argument(sx, dx.data());
+p.debug_print();
+    auto result = p.eval(pp).back();
+    std::vector<float> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+printf("result_vector has size %lu: \n", result_vector.size());
+for(float aa : result_vector) printf (" %f ", aa);printf("\n");
+ 
+    // clang-format off
+    // TODO: gold value includes floating-point rounding errors
+    std::vector<float> gold = {0.100000, 1.100000, 3.100000, 4.100000, 6.100000, 
+    9.100000, 10.100000, 12.100000, 13.100000, 15.100000, 
+    27.100000, 28.100000, 30.100000, 31.100000, 33.099998, 
+    45.099998, 46.099998, 48.099998, 49.099998, 51.099998, 
+    54.099998, 55.099998, 57.099998, 58.099998, 60.099998, 
+    72.099998, 73.099998, 75.099998, 76.099998, 78.099998};
```

Comment:
How did you get these target values? Python? Want to make sure we're comparing to something we know is right

Reply:
1.  Worked them out on paper.  The values come from the input (iota function) and the index to select is calculated.  For instance, for index (0, 2) the formula for index 2, using the half_pixel transformation and round_prefer_ceil rounding as found in gen_onnx.py, is 

(x + 0.5) / scale - 0.5 where scale for the 4th dimension is 0.601
2.5/0.601 - 0.5 = 3.659
which rounds to 4 and we take the input value at index 4.  Used the same process for the other index.
This is not as much work as it seems since the pattern is very repetitive.

2.  This needs to be replaced with a call to `verify_range` since we can't expect exact results.  The correct results should all end with .1 since that's what the iota function created.  The long decimal strings are the result of floating point rounding errors and were found by dumping program output to screen.